src/Sarif/Core/RunAutomationDetails.cs

```diff
@@ -12,8 +12,8 @@ public partial class RunAutomationDetails
 
         public bool ShouldSerializeId() => !string.IsNullOrWhiteSpace(this.Id);
 
-        public bool ShouldSerializeGuid() => !string.IsNullOrWhiteSpace(this.Guid);
+        public bool ShouldSerializeGuid() => this.Guid != null;
```

Comment:
>this.Guid [](http://example.com/codeflow?start=45&length=9)

these comparisons to null are fine, but we could also use `this.Guid.HasValue`, yes?

Reply:
yes I actually thought about this as well, I believe most people prefers `.HasValue`, 
the only reason I was thinking about consistency with how string compares with null.

Let me know your thoughts?

Below are some random points in internet:

"The compiler replaces null comparisons with a call to HasValue, so there is no real difference. Just do whichever is more readable/makes more sense to you and your colleagues."

"It's pretty silly to complain about being able to set a Nullable to null or compare it to null given that's called *Nullable*"

"I prefer (a != null) so that the syntax matches reference types."